Applications/IK/test/testIK.cpp

```diff
@@ -67,3 +73,99 @@ int main()
     cout << "Done" << endl;
     return 0;
 }
+
+void testMarkerWeightAssignments(const std::string& ikSetupFile)
+{
+    // Check the consistency for the IKTaskSet in the IKTool setup
+    InverseKinematicsTool ik(ikSetupFile);
+
+    // Get a copy of the IK tasks
+    IKTaskSet tasks = ik.getIKTaskSet();
+
+    // assign different weightings so we can verify the assignments
+    int nt = tasks.getSize();
+    for (int i{ 0 }; i < nt; ++i) {
+        tasks[i].setWeight(double(i));
+    }
+
+    cout << tasks.dump(true) << endl;
+    // update tasks used by the IK tool
+    ik.getIKTaskSet() = tasks;
+
+    // perform the check
+    checkMarkersReferenceConsistencyFromTool(ik);
+
+    // Now reverse the order and half the number of tasks
+    // so that marker and tasks lists are no longer the same
+    IKTaskSet tasks2;
+    tasks2.setName("half_markers");
+
+    for (int i{ nt / 2 }; i > -1 ; --i) {
+        tasks2.adoptAndAppend(tasks[2*i].clone());
```

Comment:
This code will segfault if `nt` is even (e.g., if `nt = 4`, then `i = 2` on the first iteration and L104 will attempt to access `tasks[4]`). I assume `nt` happens to be odd in all the examples you're using—fine for now, but it could break in the future. I suggest taking a simpler approach (and will be easier to read IMHO):
```cpp
for (int i = nt-1; i >= 0; i -= 2)
    tasks2.adoptAndAppend(tasks[i].clone());
```